compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1137,7 +1136,10 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                     // so desugar to
                     //
                     //     fn foo(x: dyn Iterator<Item = impl Debug>)
-                    ImplTraitContext::Universal(..) if self.is_in_dyn_type => (true, itctx),
+                    ImplTraitContext::Universal(_, parent) if self.is_in_dyn_type => {
```

Comment:
If parenting depends on flags like `is_in_dyn_type` unavailable in def collector, how is it possible to make it consistent?

Reply:
This specific lowering is not handled inside the def collector. The generic parameter's `DefId` is created line 1199, and its HIR parent is the item's generic parameter list.